### PR TITLE
[BUGFIX] disable ip protocol fallback and tests

### PR DESCRIPTION
**Pull request by genofire**
- fix ip protocol fallback
  - before the argument `fallbackIPProtocol` in `chooseProtocol()` (`prober/utils.go`) was never used
- correct / get default value of `IPProtocol` during testing
  - `ipv4` is a wrong value, it should be `ip4` or `ip6`
  -

### CODE REVIEW

File: prober/utils_test.go [hunk 1]
```diff
@@ -110,3 +115,30 @@ func generateTestCertificate(expiry time.Time, IPAddressSAN bool) ([]byte, []byt
 	pemKey := pem.EncodeToMemory(&pem.Block{Type: "RSA PRIVATE KEY", Bytes: x509.MarshalPKCS1PrivateKey(privatekey)})
 	return pemCert, pemKey
 }
+
+func TestChooseProtocol(t *testing.T) {
+	ctx := context.TODO()
```
Comment (brian-brazil) [inline prober/utils_test.go]: Background is fine here

File: prober/dns_test.go [hunk 2]
```diff
@@ -226,27 +230,23 @@ func TestAuthoritativeDNSResponse(t *testing.T) {
 	}{
 		{
 			config.DNSProbe{
-				IPProtocol: "ipv4",
-				QueryName:  "example.com",
+				QueryName: "example.com",
```
Comment (brian-brazil) [inline prober/dns_test.go]: Leave these as v4 please.

File: prober/dns_test.go [hunk 3]
```diff
@@ -96,21 +96,23 @@ func TestRecursiveDNSResponse(t *testing.T) {
...
+				IPProtocolFallback: true,
+				QueryName:          "example.com",
+				ValidRcodes:        []string{"SERVFAIL", "NXDOMAIN"},
 			}, false,
 		},
 		{
 			config.DNSProbe{
-				IPProtocol: "ipv4",
-				QueryName:  "example.com",
+				IPProtocol:         "ip4",
```
Comment (brian-brazil) [inline prober/dns_test.go]: Hmm, are we missing a sanity check on the value of this?

File: prober/dns_test.go (hunk 2 above)
Comment (genofire) [reply prober/dns_test.go]: with fallback?

Comment (brian-brazil) [reply prober/dns_test.go]: It doesn't matter either way.

Comment (genofire) [reply prober/dns_test.go]: thats the problem, on my notebook some test failures if there is no fallback and it try only ipv4

File: prober/dns_test.go (hunk 3 above)
Comment (genofire) [reply prober/dns_test.go]: jep, but please let somebody else do it ...

**Pull request merged**